docs/spec/spec.md

```diff
@@ -1,219 +1,166 @@
+2. Create a model definition file in persist directory
+   It will create a file with same name as the package name if no files are present in the `persist` directory. It will create an empty file with required imports(`import ballerina/persist as _;`).
+3. Create database_configurations.bal file inside the `default` module in generated directory
+   This file contains configuration variables to initialize the data stores
+    ```ballerina
+    import ballerinax/mysql.driver as _;
+
+    configurable int port = ?;
+    configurable string host = ?;
+    configurable string user = ?;
+    configurable string database = ?;
+    configurable string password = ?;
+   ```
+4. Update Ballerina.toml with database configurations.
+   It will update the Ballerina.toml file with the configurations needed for the `bal persist push` command to work.
+    ```ballerina
+    [persist.<package name>.storage.mysql]
+    host = "localhost"
+    port = 3306
+    user = "root"
+    password = ""
+    database = "" 
+   ```
+5. Create(Update) Config.toml file inside the Ballerina project.
+   It will create(update) `Config.toml` file with configurables used to initialize variables in Step 3.
+    ```ballerina
+    [<package name>]
+    host = "localhost"
+    port = 3306
+    user = "root"
+    password = ""
+    database = ""
+    ```
+
+The directory structure will be,
 ```
 medical-center
+├── generated
+         └── database_configuration.bal
+├── persist
+         └── medical-center.bal
 ├── Ballerina.toml
-├── entities.bal
+├── Config.toml
 └── main.bal
 ```
 
-Inside the Ballerina project, users can define `Entity` records as follows.
-
-```ballerina
-import ballerina/time;
-import ballerina/persist;
-
-@persist:Entity {
-    key: ["needId"],
-    tableName: "MedicalNeeds"
-}
-public type MedicalNeed record {|
-    @persist:AutoIncrement
-    readonly int needId = -1;
-
-    int itemId;
-    int beneficiaryId;
-    time:Civil period;
-    string urgency;
-    int quantity;
-|};
-```
-
-In this example, `MedicalNeed` is an entity with the attributes `itemId`, `beneficiaryId`, `period`, `urgency`, and `quantity`. When users execute `bal persist generate` inside a Ballerina project, a new module named `clients` will be created. Inside the `clients` module, Ballerina files will be generated encapsulating the client object with respect to each entity defined in the project. The following is the project structure after the command execution.
-```
-medical-center
-├── Ballerina.toml
-├── entities.bal
-├── main.bal
-└── modules
-    └── clients
-        └── medicalneed_client.bal
-```
+Behaviour of the `init` command,
+- User should invoke the command within a bal project
+- If the user invokes the command twice, it will not fail. It will execute the steps once again.
```

Comment:
```suggestion
- If the user invokes the command twice, it will not fail. It will verify that all the configurations are in place for the definition files defined inside the `persist` directory. If not, add missing configurations and files.
```